cocotb/regression.py

```diff
@@ -75,35 +79,18 @@ def _my_import(name):
 class RegressionManager:
     """Encapsulates all regression capability into a single place"""
 
-    def __init__(self, root_name, modules, tests=None, seed=None, hooks=[]):
+    def __init__(self, dut: SimHandle):
```

Comment:
It's tempting to split this in two, as:
```
def __init__(self, dut: SimHandle, tests: List[Test], hooks: List[Hook]):
   ...

@classmethod
def from_env_vars(cls, dut: SimHandle):
    # possibly just inline these methods
    tests = cls.discover_tests()
    hooks = cls.discover_hooks()
    return cls(dut, tests=tests, hooks=hooks)
```

Fine as a follow-up PR. One complexity is what logger to use before the manager is instantiated.

Reply:
It almost seems like instead of moving the discovery code into the RM, we should move it out and not have the named constructor. At that point we would use the root logger.

Reply:
Or just use `SimLog("cocotb.regression")` in both places anyway, since that's a global logger instance.